commons/com.b2international.index.api/src/com/b2international/index/revision/DefaultRevisionIndex.java

```diff
@@ -109,65 +125,64 @@ private RevisionCompare compare(final RevisionBranch base, final RevisionBranch
 			public RevisionCompare execute(Searcher searcher) throws IOException {
 				final Set<Integer> commonPath = Sets.intersection(compare.segments(), base.segments());
 				final Set<Integer> segmentsToCompare = Sets.difference(compare.segments(), base.segments());
+				final RevisionBranch baseOfCompareBranch = new RevisionBranch(base.path(), Ordering.natural().max(commonPath), commonPath);
 				
 				final Set<Class<? extends Revision>> typesToCompare = getRevisionTypes();
-				final Builder result = RevisionCompare.builder(DefaultRevisionIndex.this, new RevisionBranch(base.path(), Ordering.natural().max(commonPath), commonPath), compare);
+				final Builder result = RevisionCompare.builder(DefaultRevisionIndex.this, baseOfCompareBranch, compare);
 				
-				final Map<Class<? extends Revision>, LongSet> newAndChangedComponents = newHashMap();
-				final Map<Class<? extends Revision>, LongSet> deletedAndChangedComponents = newHashMap();
+				final Multimap<Class<? extends Revision>, Revision.Views.StorageKeyAndHash> newAndChangedComponents = ArrayListMultimap.create();
+				final Multimap<Class<? extends Revision>, Revision.Views.StorageKeyAndHash> deletedAndChangedComponents = ArrayListMultimap.create();
 				
 				// query all registered revision types for new, changed and deleted components
 				for (Class<? extends Revision> typeToCompare : typesToCompare) {
-					final Query<Revision.Views.StorageKeyOnly> newAndChangedQuery = Query
-							.selectPartial(Revision.Views.StorageKeyOnly.class, typeToCompare)
+					final Query<Revision.Views.StorageKeyAndHash> newAndChangedQuery = Query
```

Comment:
Makes sense. I'll use `newOrChanged` then, because a revision can not be `new` and `changed` at the same time.